pf/internal/schemashim/type_schema.go

```diff
@@ -66,6 +66,9 @@ func (s *typeSchema) Elem() interface{} {
 	case types.MapType:
 		contract.Assert(s.nested == nil || len(s.nested) == 0)
 		return newTypeSchema(tt.ElemType, nil)
+	case pfattr.TypeWithElementTypes:
```

Comment:
This might match more than tuples, it might match set.; perhaps lift the case types.SetType into a separate match.

Reply:
It shouldn't match `set`. According to the documentation, set should implement `pfattr.TypeWithElementType` (sets, lists and map), but not `pfattr.TypeWithElementTypes` (tuple). Note the plural difference.

Reply:
Aha!